app/javascript/src/components/Dashboard/Contacts/NewContactForm.jsx

```diff
@@ -1,7 +1,69 @@
 import React from "react";
+import * as yup from "yup";
+import { Formik, Form } from "formik";
+import { Input, Select, Switch } from "neetoui/formik";
+import { Button, Label } from "neetoui";
+import { departmentOptions } from "constants/contacts";
 
-const NewContactForm = () => {
-  return <div>contact form</div>;
-};
-
-export default NewContactForm;
+export default function NewContactForm({ onClose, refetch }) {
+  const handleSubmit = () => {
+    refetch();
+    onClose();
+  };
+  return (
+    <Formik
+      initialValues={{
+        name: "",
+        email: "",
+        department: {},
+        contactNumber: "",
+        addedToBasecamp: false,
+      }}
+      onSubmit={handleSubmit}
+      validationSchema={yup.object({
+        name: yup.string().required("Name is required"),
+        email: yup
+          .string()
+          .email("Invalid email format")
+          .required("Email is required"),
+        contactNumber: yup.string().required("Contact Number is required"),
+      })}
+    >
+      {({ isSubmitting }) => (
+        <Form>
+          <Input label="Name" name="name" className="mb-6" />
+          <Input label="Email" name="email" className="mb-6" />
+          <Input label="Contact Number" name="contactNumber" className="mb-6" />
+          <Select
+            label="Department"
+            placeholder="Select a department"
+            name="department"
+            options={departmentOptions}
+            className="mb-6"
+          />
+          <div className="flex justify-between mb-6">
+            <Label>Add to Basecamp</Label>
+            <Switch name="addedToBasecamp" />
+          </div>
```

Comment:
Instead of using `mb-6` for each of the blocks individually, please use `space-y-6`.

Please refer [space between](https://tailwindcss.com/docs/space#class-reference) from tailwindCSS for more details.